htdocs/comm/action/index.php

```diff
@@ -919,7 +919,7 @@
 // Complete $eventarray with events coming from external module
 $parameters=array(); $object=null;
 $reshook=$hookmanager->executeHooks('getCalendarEvents',$parameters,$object,$action);
-if (! empty($hookmanager->resArray['eventarray'])) $eventarray=array_merge($eventarray, $hookmanager->resArray['eventarray']);
+if (! empty($hookmanager->resArray['eventarray'])) $eventarray=array_replace($eventarray, $hookmanager->resArray['eventarray']);
```

Comment:
Currently, no existing module can work correclty (we may have illusion it works with array_replace) but it's just an illusion (some data may be lost). This is enough to justify a regression for external module. So, can you push an implementation like your first idea (with a loop to complete the array) ?